src/index.js

```diff
@@ -1,73 +1,72 @@
+import { enableTaskEditing, saveTaskEdit } from './modules/editFunctions.js';
 
 const taskForm = document.getElementById('task-form');
 const taskInput = document.getElementById('task-input');
 const taskList = document.getElementById('task-list');
 const deleteSelectedButton = document.getElementById('delete-selected');
 
-const hardcodedTasks = [
-  { text: 'Buy Groceries', completed: false },
-  { text: 'Complete ToDo List Project', completed: true },
-  { text: 'Do coding challenges', completed: false },
-];
-
-function saveTasksToLocalStorage() {
-  const tasks = [];
+// Save data to local storage on page load
+const savedTasks = JSON.parse(localStorage.getItem('tasks')) || [];
+savedTasks.forEach((savedTask) => {
+  addTaskToList(taskList, savedTask.text);
   const taskItems = taskList.querySelectorAll('li');
-  taskItems.forEach((taskItem) => {
-    const checkbox = taskItem.querySelector("input[type='checkbox']");
-    const text = taskItem.querySelector('span').textContent;
-    const completed = checkbox.checked;
-    tasks.push({ text, completed });
-  });
-  localStorage.setItem('tasks', JSON.stringify(tasks));
-}
-
-function displayTasks() {
-  hardcodedTasks.forEach((task) => {
-    const li = document.createElement('li');
-    li.innerHTML = `
-      <input type="checkbox" ${task.completed ? 'checked' : ''}>
-      <span>${task.text}</span>
-    `;
-    taskList.appendChild(li);
-  });
-}
-
-displayTasks();
+  const checkbox = taskItems[taskItems.length - 1].querySelector("input[type='checkbox']");
+  if (savedTask.completed) {
+    checkbox.checked = true;
+    updateTaskTextDecoration(checkbox, savedTask.text);
+  }
+});
 
 taskForm.addEventListener('submit', (event) => {
   event.preventDefault();
 
   const taskText = taskInput.value.trim();
   if (taskText === '') return;
 
-  const li = document.createElement('li');
-  li.innerHTML = `
-    <input type="checkbox">
-    <span>${taskText}</span>
-  `;
-  taskList.appendChild(li);
+  addTaskToList(taskList, taskText);
 
   taskInput.value = '';
 
-  saveTasksToLocalStorage();
+  saveTasksToLocalStorage(taskList);
 });
 
 taskList.addEventListener('change', (event) => {
   if (event.target.tagName === 'INPUT' && event.target.type === 'checkbox') {
     const taskText = event.target.nextElementSibling;
-    taskText.style.textDecoration = event.target.checked ? 'line-through' : 'none';
-    saveTasksToLocalStorage();
+    updateTaskTextDecoration(event.target, taskText);
+    saveTasksToLocalStorage(taskList);
   }
 });
 
 deleteSelectedButton.addEventListener('click', () => {
-  const selectedCheckboxes = taskList.querySelectorAll('input[type="checkbox"]:checked');
-  selectedCheckboxes.forEach((checkbox) => {
-    const listItem = checkbox.parentElement;
-    taskList.removeChild(listItem);
-  });
+  deleteSelectedTasks(taskList);
 
-  saveTasksToLocalStorage();
+  // After deleting, update indexes
+  updateIndexes(taskList);
+  saveTasksToLocalStorage(taskList);
+});
+
```

Comment:
| @DevKaranJ Design |
|-|
| ![image](https://github.com/DevKaranJ/ToDo-list/assets/93039359/1594c8e0-13fc-4931-a767-85d1547c7063) |

| Project Requirement |
|-|
| ![image](https://github.com/DevKaranJ/ToDo-list/assets/93039359/29e4da35-657e-4326-b6ce-ac935aad9aea) |

- [ ] It would be nice if you double-check the requirements because by default you should have three key-pair values but you are missing the index 😨 Kindly add it into your app as expected in the project requirements 😊 👍
- [ ] It would be also nice when you will remove a todo from a list then the index will be reset according to the project requirements 😊

Reply:
- [ ] Project requirements.
- As requested by the previous reviewer, when you add the task, it should also add a property `index` to the task object. 
![image](https://github.com/DevKaranJ/ToDo-list/assets/33268927/7433fd1e-2ee3-49d4-a325-8a65099e9dd7)
- Also make sure, that when you delete the task, it updates the indices of all the tasks as per the project requirements.